nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/AbstractAzureDataLakeStorageProcessor.java

```diff
@@ -0,0 +1,154 @@
+        .name("storage-account-key").displayName("Storage Account Key")
+        .description("The storage account key. This is an admin-like password providing access to every container in this account. It is recommended " +
+            "one uses Shared Access Signature (SAS) token instead for fine-grained control with policies. " +
+            "There are certain risks in allowing the account key to be stored as a flowfile " +
+            "attribute. While it does provide for a more flexible flow by allowing the account key to " +
+            "be fetched dynamically from a flow file attribute, care must be taken to restrict access to " +
+            "the event provenance data (e.g. by strictly controlling the policies governing provenance for this Processor). " +
+            "In addition, the provenance repositories may be put on encrypted disk partitions.")
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .required(false)
+        .sensitive(true).build();
+
+    public static final PropertyDescriptor ACCOUNT_NAME = new PropertyDescriptor.Builder()
+        .name("storage-account-name").displayName("Storage Account Name")
+        .description("The storage account name.  There are certain risks in allowing the account name to be stored as a flowfile " +
+            "attribute. While it does provide for a more flexible flow by allowing the account name to " +
+            "be fetched dynamically from a flowfile attribute, care must be taken to restrict access to " +
+            "the event provenance data (e.g. by strictly controlling the policies governing provenance for this Processor). " +
+            "In addition, the provenance repositories may be put on encrypted disk partitions." +
+            " Instead of defining the Storage Account Name, Storage Account Key and SAS Token properties directly on the processor, " +
+            "the preferred way is to configure them through a controller service")
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .required(true)
+        .sensitive(true).build();
+
+    public static final PropertyDescriptor FILESYSTEM = new PropertyDescriptor.Builder().name("Filesystem-name")
+        .displayName("Filesystem Name")
+        .description("Name of the Azure Storage File System, It is assumed to be already existing")
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .required(true)
+        .build();
+
+    public static final PropertyDescriptor DIRECTORY = new PropertyDescriptor.Builder().name("Directory-name")
+        .displayName("Directory Name")
+        .description("Name of the Azure Storage Directory, It is assumed to be already existing")

```

Comment:
I think the non existing directories will be created. At least I did not get any errors.